DatadogInternal/Sources/MessageBus/FeatureMessage.swift

```diff
@@ -8,13 +8,24 @@ import Foundation
 
 /// The set of messages that can be transimtted on the Features message bus.
 public enum FeatureMessage {
-    /// A custom message with generic encodable
-    /// attributes.
+    /// A custom message with generic encodable attributes.
+    ///
+    /// A baggage can be used to transmit loosely-typed data structure using `Codable`.
+    /// The encoding/decoding processes will have an impact on performances, opt for a baggage
+    /// only if the data-structure is small.
+    ///
+    /// For large data type, use the `.value` case with shared type definition.
     case baggage(
         key: String,
         baggage: FeatureBaggage
     )
 
+    /// A message value.
+    ///
+    /// Use this message case when the value type is known end-to-end.
+    /// Such type can be defined as part of the `DatadogInternal` module.
+    case value(Any)
```

Comment:
We could also use a case per type, such as `case webview(WebViewMessage)`. But it will be hardly scalable as we increase the number of types to transmit.

Reply:
It would show a clean list of supported types in the contract though. How many different models do we anticipate at this point?

Reply:
I'd be pretty much in favour of:
```swift
case webview(WebViewMessage)
```

The main benefit is being explicit and type-safe on both ends. An extra thing is having a clear definition point for documenting all messages that features can exchange.

> But it will be hardly scalable as we increase the number of types to transmit.

Could you elaborate? IMO with the growing number of types that message bus transmits, having clarity on what exact type both sides expect sounds critical. I don't see this against scalability.

I'd chose the simplest, typed solution to start with for maximum safety. We could see how it grows and adjust.

Reply:
Indeed, I'm probably anticipating too much here. We should go with an explicit set of messages, I will change that 👍 

Here a list of message that would need type sharing:
- error message from log
- rum view
- rum session state
- crash report
- web-view

Reply:
And that would be perfectly fine to have them all listed 👍. Long-term we can see what would be the cost of maintaining this in implementation **and tests**. Especially in tests, I can foresee the [FeatureMessageMocks.swift](https://github.com/DataDog/dd-sdk-ios/blob/2.7.1/TestUtilities/Mocks/FeatureMessageMocks.swift) constantly growing with new `first*()` and `as*` helpers, which is what concerns me.

Reply:
Done, I have replaced the `.value` with `. webview` 👍